tests/conditional_formatting/conditional_formatting_plugin.test.ts

```diff
@@ -679,6 +680,278 @@ describe("conditional format", () => {
     });
   });
 
+  describe("Grid Manipulation Formulas", () => {
+    function cellIsRuleEqual(formula: string): ConditionalFormatRule {
+      return {
+        type: "CellIsRule",
+        values: [formula],
+        operator: "Equal",
+        style: { fillColor: "orange" },
+      };
+    }
+
+    function cellIsRuleBetween(formula: string): ConditionalFormatRule {
+      return {
+        type: "CellIsRule",
+        values: [formula, formula],
+        operator: "Between",
+        style: { fillColor: "orange" },
+      };
+    }
+
+    function colorScaleRule(formula: string): ConditionalFormatRule {
+      return {
+        type: "ColorScaleRule",
+        minimum: {
+          type: "formula",
+          value: formula,
+          color: 0,
+        },
+        maximum: {
+          type: "formula",
+          value: formula,
+          color: 0,
+        },
+        midpoint: {
+          type: "formula",
+          value: formula,
+          color: 0,
+        },
+      };
+    }
+
+    function iconSetRule(formula: string): ConditionalFormatRule {
+      return {
+        type: "IconSetRule",
+        icons: {
+          upper: "u",
+          middle: "m",
+          lower: "l",
+        },
+        upperInflectionPoint: {
+          type: "formula",
+          operator: "gt",
+          value: formula,
+        },
+        lowerInflectionPoint: {
+          type: "formula",
+          operator: "gt",
+          value: formula,
+        },
+      };
+    }
+
+    function formulasToCFs(
+      formulas: string[],
+      cfrule: (formula: string) => ConditionalFormatRule
+    ): ConditionalFormat[] {
+      return formulas
+        .map((formula) => cfrule(formula))
+        .map((rule) => {
+          return {
+            id: "1",
+            rule,
+            ranges: ["A1"],
+          };
+        });
+    }
+
+    const cfGenerators = [
+      ["cellIsRuleEqual", cellIsRuleEqual],
+      ["cellIsRuleBetween", cellIsRuleBetween],
+      ["colorScaleRule", colorScaleRule],
+      ["IconSetRule", iconSetRule],
+    ] as [string, (s: string) => ConditionalFormatRule][];
```

Comment:
```suggestion
    ] as const;
```
Slightly easier and shorter and works as well